src/main/resources/org/broadinstitute/hellbender/utils/config/GATKConfig.properties

```diff
@@ -0,0 +1,65 @@
+# ======================================================================================================================
+# System Properties
+#
+# All Configuration options in ths section are to be injected into the JRE
+# as System Properties.
+#
+# Any property beginning with `system.` (without quotes) will be injected into the JRE as a System Property.
+# Properties not starting with `system.` (without quotes) will NOT be used as System Properties.
+#=======================================================================================================================
+
+# ---------------------------------------------------------------------------------
+# General Options:
+# ---------------------------------------------------------------------------------
+
+gatk_stacktrace_on_user_exception = true
+
+# ---------------------------------------------------------------------------------
+# SAMJDK Options:
+# ---------------------------------------------------------------------------------
+
+system.samjdk.use_async_io_read_samtools = true
+system.samjdk.use_async_io_write_samtools = true
+system.samjdk.use_async_io_write_tribble = false
+system.samjdk.compression_level = 1
+
+# ----------------------------------------------------------
+# Spark Options:
+# ----------------------------------------------------------
+
+spark.kryoserializer.buffer.max = 512m
+spark.io.compression.codec = lzf
+spark.yarn.executor.memoryOverhead = 600
+
+spark.driver.maxResultSize = 0
+spark.driver.userClassPathFirst = true
+spark.driver.extraJavaOptions =
+
+spark.executor.extraJavaOptions =
+
+# ---------------------------------------------------------------------------------
+# Other Options:
+# ---------------------------------------------------------------------------------
+
+snappy.disable = true
+
+# ======================================================================================================================
+# Tool Configuration File
+#
+# Contains config options for various pieces of the GATK
+# ======================================================================================================================
+
+# ---------------------------------------------------------------------------------
+# General Options:
+# ---------------------------------------------------------------------------------
+
+codec_packages = htsjdk.variant      ,            htsjdk.tribble         ,         org.broadinstitute.hellbender.utils.codecs          
```

Comment:
@magicDGS with a user-provided file the user can specify whatever option for `codec_packages` they want.  That way they can remove those three libraries if they need to.  The values here are packaged in the JAR file so that if there is no user config file, the GATK can still run off a default set of options (it will also fall back to hard defaults specified in the `DefaultValue` annotation in `GATKConfig.java` if this file is somehow deleted).

Why couldn't this be solved with a new config option `codec_excludes` which is a list of string values for the codecs to exclude from the run?